### PR TITLE
10.0 refactor py3o jne

### PR DESCRIPTION
**Pull request by JonathanNEMRY**
This PR applies same changes than https://github.com/faide/reporting-engine/pull/14 

cc @alexis-via

### CODE REVIEW

**Comment by lmignon** [comment]
@alexis-via @JonathanNEMRY I'll refactor a little bit more to avoid too much memory consumption.. 
I'll update the 2 PRS once done (V9 and V10)

File: report_py3o/models/py3o_report.py [hunk 1]
```diff
@@ -181,40 +208,87 @@ def create_single_pdf(self, cr, uid, ids, data, report_xml, context=None):
...
+                prefix='py3o-template-'
             ) as fd:
                 for chunk in r.iter_content(chunk_size):
                     fd.write(chunk)
                 fd.seek(0)
                 # ... but odoo wants the whole data in memory anyways :)
                 res = fd.read()
+        self._postprocess_report(
+            res, model_instance.id, save_in_attachment)
+        return res, "." + self.ir_actions_report_xml_id.py3o_filetype
```
Comment (alexis-via) [inline report_py3o/models/py3o_report.py]: I fear you re-introduced the "double dot in filename" bug, that I fixed with this commit https://github.com/akretion/reporting-engine/commit/b702d15a2982f0075c1cacd7a0c66b2138372866

Did you test it this PR and check the filenames of the py3o reports generated ?

**Comment by alexis-via** [comment]
@lmignon OK. So I understand I should wait for more commits from you guys before merging this PR. Good luck for your optimisations.

**Comment by lmignon** [comment]
@alexis-via Yes you are right. The optimisation is to always use files in the generation process and only at the end load the file into memory since it's what odoo expect... In the same time it become possible to override the postprocess method if you need to add a digital signature to your generated document for exemple. The posprocess mehod will be proposed to odoo on 'report' and if accepted by odoo we finally have a point of extension also for the generated qweb reports. You can also see that when the report is launched on a set of records I return a merged pdf if the result is pdf or a zip file with all the generated reports...

**Comment by lmignon** [comment]
@JonathanNEMRY @alexis-via My improvements are there... https://github.com/faide/reporting-engine/pull/14/commits/9befca91d45ac09416a4a1d680089bfa1c0c308c
These should be tested (have done some but not enough)...
runbot is green

**Comment by alexis-via** [comment]
Just drop me a note when I can merge. I'll have several go-live on 1st January with report_py3o, so if we merge this before Jan 1st, it will be a good "live test" :)

**Comment by alexis-via** [comment]
@lmignon  tell me when you want me to merge this.

**Comment by faide** [comment]
@lmignon sorry for the ping on the Wrong PR. Can we merge this one in the main PR ? or do you need more time on this ?

**Comment by lmignon** [comment]
@faide yes you can!
@alexis-via This one is ready to merge :smirk:

**Comment by lmignon** [comment]
Merci @alexis-via !

**Pull request merged**